internal/controllers/topology/cluster/structuredmerge/options.go

```diff
@@ -33,10 +33,11 @@ var (
 		{"metadata", "namespace"},
 		// uid is optional for a server side apply intent but sets the expectation of an object getting created or a specific one updated.
 		{"metadata", "uid"},
-		// the topology controller controls/has an opinion for labels, annotation, ownerReferences and spec only.
+		// the topology controller controls/has an opinion for labels, annotation, ownerReferences, finalizers and spec only.
 		{"metadata", "labels"},
 		{"metadata", "annotations"},
 		{"metadata", "ownerReferences"},
+		{"metadata", "finalizers"},
```

Comment:
I would like to drop adding the finalizer in desired_state.go (l.626-631).
I think we shouldn't  add the finalizer on the first reconcile and then drop it in every subsequent one as we know have a stable solution in the MD topology controller.


No strong opinions on if we shoud leave finalizers here after dropping the code in desired_state.go. In other cases we only added paths to the list that we actually want to set somewhere.

Reply:
Let's wait to see what Fabrizio says

Reply:
EDIT: Also fine for me, let's see :)

Reply:
As discussed the new version only adds the finalizer in the MachineDeployment and MachineSet topology controllers. Previously the finalizer was still added on creation.

Reply:
Sounds good will take a look tomorrow